### DISCUSSION TITLE
Component-level caching?

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#17731] [Help]
Most of my pages are SSG, using `getStaticProps` I've managed to get a lot of work done, but time has come to update some of my components with data that comes from APIs.

The problem is - all pages are static, very few components require an API call for data, so, in essence, the pages aren't really static. I've read on `swr` but still can't make sense of the strategy I need to employ. If it was just a page, then I could just simply cache these API calls and serve them with `getStaticProps` and call it a day, but since it's a component, I can't use that.

What should I do? `getServerSideProps` isn't really useful here, nor desired. This data doesn't need to update **on each request, just every now and then and it's the same for every user.**

---

Note 1: I could just `swr` inside my component, but that's an extra call to the server. I could, on the server, install `redis` and cache the API calls' results and just simply serve it from redis.

**Comment by jamesmosier** [comment]
> but since it's a component, I can't use that.

You could make all the API calls you need in `getStaticProps` and then pass the resulting data/props to the components that need them from your page. All of the components would be children of your page, so you can just hydrate them with the data they need.
